src/service/src/grpc.rs

```diff
@@ -376,9 +374,40 @@ where
                     _ = &mut cancel_rx => break,
                 }
             }
-            info!("GrpcServer: remote client disconnected");
         };
-        Ok(Response::new(Box::pin(response)))
+        Ok(Response::new(ResponseStream::new(stream)))
+    }
+}
+
+/// A stream returning responses to GRPC clients.
+///
+/// This is defined as a struct, rather than a type alias, so that we can define a `Drop` impl that
+/// logs stream termination.
+pub struct ResponseStream<PR>(Pin<Box<dyn Stream<Item = Result<PR, Status>> + Send>>);
```

Comment:
Thanks! There are a couple issues with this approach I noticed:

* It makes `ResponseStream` generic over the stream, not the response type, so other existing code like this stops working: `type CommandResponseStreamStream = ResponseStream<ProtoStorageResponse>;`. We can of course rewrite it but it does become more cumbersome in the process.
* (minor): We need a new dependency for this. Not a new dependency to Mz, just to the `mz-service` crate.
* (minor): Since we have a `Drop` impl, we also need to switch to `PinnedDrop`.

In the end I think the resulting code isn't more readable than what exists now. It avoids an allocation, but this doesn't seem like a huge win, given that this type is not used on a hot path. So I'm going to leave it as is for now.